src/Tokenizer.php

```diff
@@ -314,47 +336,47 @@ final class Tokenizer
      * @var list<string>
      */
     private array $reservedToplevel = [
-        'WITH',
-        'SELECT',
-        'FROM',
-        'WHERE',
-        'SET',
-        'ORDER BY',
```

Comment:
i'm not really sure about this change.... `ORDER BY` is a keyword. `ORDER` by it self in most of sql dialects means nothing. (same for all the other "composite keywords we had").

If we were to say that keywords are uppercase and just "single words", this library would be much simpler....

Reply:
Here is MySQL spec - https://dev.mysql.com/doc/refman/8.4/en/keywords.html

`ORDER` and `BY` are "reserved" keywords and must be specified separately.

"reserved" keywords are special as they cannot be used in SQL queries without quotes - ex: https://dbfiddle.uk/_IuDbiFR

This is also why we have the `testKeywordsReservedAreUpperCasedAndWithoutWhitespace` test :)

Reply:
how would  the query `select 'x' AS order` be formatted by this library before and after your changes?

Reply:
Perfectly, as it is invalid query as shown in the online repro :)

Reply:
At first, it is invalid query.

But to make you satisfied, it will be formatted exactly the same as before. This PR fixed "reserved" keywords. "reservedToplevel" "possible multi-word keywords" are the same as before.

Reserved keywords needs to match the SQL spec to be able to rely they needs to be quoted - and prevent bugs like #95.

Reply:
@goetas @derrabus does it make sense to you and can I get your approval so I can continue with #106 where the "reserved" keywords will be extended and fixed to match strictly the spec?

Reply:
Could you please answer to https://github.com/doctrine/sql-formatter/pull/107#discussion_r1609804049?

Reply:
@goetas answered in https://github.com/doctrine/sql-formatter/pull/107#discussion_r1609812253 - the answer is exactly the same as before.